Purchases/Identity/CustomerInfo.swift

```diff
@@ -216,6 +216,21 @@ import Foundation
         self.allPurchases = subscriberData.allPurchases
     }
 
+    static func from(json maybeJSON: [String: Any]?) throws -> CustomerInfo {
+        guard let customerJSON = maybeJSON else {
+            throw UnexpectedBackendResponseSubErrorCode.customerInfoResponseMalformed
```

Comment:
hmm we _could_ be more specific here that it's nil, not just malformed? but i'll review on to see where this is called from...

Reply:
This is a carry over from old code. Any issue was considered malformed. I could rename this to nil, but malformed is, technically correct too 😄
Let me know if you think it should be renamed.

Reply:
i guess it depends on how often this scenario happens. we could do in a follow-up PR too, i know it's not part of this work. i just know how much we've been moving towards more specific errors so thought i'd call it out!

Reply:
✅ updated.